### PR TITLE
Add examples for attributes

### PR DESCRIPTION
**Pull request by mikekistler**
# Add examples for attributes

This PR just adds remarks and examples to the XML doc for some commonly used attributes.

## Description

Add remarks and examples to the XML doc for some commonly used attributes in Web apps to improve the documentation that appears on Learn and in VS / VSCode.

The intent is to add doc for:
- [x] FromBody
- [x] FromForm
- [x] FromHeader
- [x] FromQuery
- [x] FromRoute
- [ ] ~Produces~
- [ ] ~Produces\<T>~
- [ ] ~ProducesResponseType~
- [ ] ~ProducesResponseType\<T>~

and maybe a few more.

This will help developers understand some quirks about the way attributes can be used, such as using named parameters to set properties of the attribute even when these are not present in any constructor.

Update: Scoped this PR to just the attributes for parameter binding.

### CODE REVIEW

**Comment by mikekistler** [comment]
I think this is ready for review now. I'll follow up with PRs for other attributes.

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs [hunk 1]
```diff
@@ -10,9 +10,21 @@ namespace Microsoft.AspNetCore.Mvc;
 /// Specifies that a parameter or property should be bound using the request body.
 /// </summary>
 /// <remarks>
+/// Binds a parameter or property to the request body.
+///
 /// By default, ASP.NET Core MVC delegates the responsibility of reading the body to an input formatter.<br/>
 /// In the case of ASP.NET Core Minimal APIs, the body is deserialized by <see cref="System.Text.Json.JsonSerializer"/>.
+///
+/// For more information about parameter binding see
+/// <see href="https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis/parameter-binding?">Parameter binding</see>.
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:19]:
```suggestion
/// <see href="https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis/parameter-binding">Parameter binding</see>.
```

Is the trailing `?` here and elsewhere intentional?

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs [hunk 2]
```diff
@@ -10,9 +10,21 @@ namespace Microsoft.AspNetCore.Mvc;
...
 /// </remarks>
+/// <example>
+/// In this example, the value of the 'name' field in the form-data request body is bound to the name parameter,
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:22]: It's a little hard to understand what is meant here when there is no `Person` type defined in the example code? Maybe we can add a record definition to make it clearer?

File: src/Mvc/Mvc.Core/src/FromRouteAttribute.cs [hunk 3]
```diff
@@ -10,6 +10,20 @@ namespace Microsoft.AspNetCore.Mvc;
...
+/// or the name specified in the <see cref="Name"/> property.
+/// The route parameter name is matched against parameter segments of the route pattern case-insensitively.
+///
+/// For more information about parameter binding see
+/// <see href="https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis/parameter-binding?">Parameter binding</see>.
+/// </remarks>
+/// <example>
+/// In this example, the value of the 'id' route parameter is bound to the parameter.
+/// <code>
+/// app.MapGet("/from-route/{id}", ([FromRoute] string id) => id);
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/src/FromRouteAttribute.cs:24]: What happens if you pass a name argument that doesn't match the route template, like `[FromRoute(Name = "notId")]`? Should we document this in the docstring?

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs [hunk 4]
```diff
@@ -10,9 +10,21 @@ namespace Microsoft.AspNetCore.Mvc;
...
+/// and the value of the 'age' field is bound to the age parameter.
+/// <code>
+/// app.MapPost("/from-body", ([FromBody] Person person) => person);
+/// </code>
+/// </example>
 [AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
 public class FromBodyAttribute : Attribute, IBindingSourceMetadata, IConfigureEmptyBodyBehavior, IFromBodyMetadata
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:29]: Should we document something about `EmptyBodyBehavior` here?

**Review by captainsafia — COMMENTED** [review]
Nice stuff! Left some comments inline.

File: src/Mvc/Mvc.Core/src/FromBodyAttribute.cs (hunk 1 above)
Comment (mikekistler) [reply src/Mvc/Mvc.Core/src/FromBodyAttribute.cs:19]: Oh good catch! I will fix.

File: src/Mvc/Mvc.Core/src/FromRouteAttribute.cs (hunk 3 above)
Comment (mikekistler) [reply src/Mvc/Mvc.Core/src/FromRouteAttribute.cs:24]: If the name does not match a parameter in the route template, the build will fail, e.g.:
```
/home/vscode/.nuget/packages/microsoft.extensions.apidescription.server/9.0.0-rc.2.24474.3/build/Microsoft.Extensions.ApiDescription.Server.targets(68,5): error : System.AggregateException: One or more errors occurred. ('notId' is not a route parameter.)
```

I'm not sure this is worth mentioning but happy to add it if you feel strongly about it.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**